src/librustc_typeck/check/mod.rs

```diff
@@ -3344,6 +3344,12 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
                                 }
                             };
                     }
+                    ty::Array(ty, _) if ty.is_numeric() => {
```

Comment:
Is there any reason to limit this suggestion to only `[{integer}; _]` and `[{float}; _]`? I'm not against it, as being conservative in suggestions is not bad, I just want to hear what was _your_ reasoning.